addons/mail/static/src/models/message/message.js

```diff
@@ -442,6 +450,12 @@ function factory(dependencies) {
     }
 
     Message.fields = {
+        anonymousName: attr({
+            compute: '_computeAnonymousName',
+            dependencies: [
+                'originThread',
```

Comment:
It depends also on `originThread.anonymous_name`.

Unfortunately relations are not yet supported in `dependecies`, so you'll have to create an intermediary related field such as `originThreadAnonymousName` that has `related: 'originThread.anonymous_name'`.

(example `messagingCurrentPartner` in message model)